### PR TITLE
Yakir 1st draft of exec1

### PR DESCRIPTION
**Pull request by yakirch**

### CODE REVIEW

**Comment by yakirch** [comment]
@juliavol

**Comment by yakirch** [comment]
test

File: home-assignments/session1/age.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+def pairwise(iterable):
+    it = iter(iterable)
+    a = next(it, None)
+
+    for b in it:
+        yield (a, b)
+        a = b
+
+
+def main():
```
Comment (Juliavol) [inline home-assignments/session1/age.py]: We were requested to break the code into separate functions.
Please consider separating the functionality to outside functions to be called by the main

File: home-assignments/session1/age.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        age_grp = my_content['buckets']
+        age_grp.sort()
+        age_young = min(age_grp)
+        age_old = max(age_grp)
+        # Generator that return the age with the next age in the age list
+        gen_ages = pairwise(age_grp)
+        # Preparing a dict with list as values for the yaml dump
+        yaml_age_per_grp = defaultdict(list)
+        #yaml_age_per_grp = {}
+        print (yaml_age_per_grp)
```
Comment (Juliavol) [inline home-assignments/session1/age.py]: Please consider removing unnecessary prints - good for debugging, bad for production ;) )

File: home-assignments/session1/age.py [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+            if val <= age_young:
+                yaml_age_per_grp[str('<'+str(age_young))].append(key)
+            if val >= age_old:
+                yaml_age_per_grp[str('>' + str(age_old))].append(key)
+
+        print(yaml_age_per_grp)
+        #print(yaml_age_per_grp)
+        print(yaml.dump(yaml_age_per_grp, default_flow_style=False))
+    with open('outfile.yaml',mode='wt') as yaml_output_file:
+        yaml.dump(yaml_age_per_grp,yaml_output_file,default_flow_style='false')
```
Comment (Juliavol) [inline home-assignments/session1/age.py]: Lines 59 and 61: Please use a single convention - either `False` or `'false'`

All in all - loogs good :)

**Pull request closed without merging**